### PR TITLE
Add collapsible table logic

### LINKED ISSUE #73: Make run entries a fixed width
### Description of the issue

Currently runs in the reduction history page are of varying widths. This is a bit jarring when browsing. Make it so the widths of each run are the same.

### LINKED ISSUE #104: Add sticky headers to the table in reduction history
https://mui.com/material-ui/react-table/#sticky-header

### LINKED ISSUE #105: Replace reduction history table with a collapsible table that provides multiple things
Edit the reduction history table to use a collapsible table (https://mui.com/material-ui/react-table/#collapsible-table)

Each row would need to show when it opens:
- A button to navigate to the data view, think the instrument page
- The status message which will either be nothing or an error or a stacktrace (need to be able to handle large text, and no text perhaps by itself on the right hand side?)
- The status again, but in text this time.

Not sure if any other info needs to appear here, I welcome thoughts and feedback.

--------------
With h5 viewer in mind:

Taking inspiration from here:

Image

The left side is a list of outputs, (max length shown is like 5), but can be scrolled down. Replaced by status message and stacktrace on an error occurring. 

Right hand side is the info we don't have on the row already perhaps also moving some things that make sense like reduction start and end times. Then a button for a "value editor" and "re run" where value editor opens a new tab that contains a monaco script editor and a set of tools to do reruns of reductions/data processing (Details to be confirmed) the re-run just re-runs the reduction with the same settings in case of a flaky reduction.

### LINKED ISSUE #106: Move status to it's own column out of the experiment number
Add a new column between the experiment number and run start columns, then move the status there. The status should also no longer have the status message in the tooltip and show the text of the actual status, such as success, running (assume not_started is running for the purpose of user facing text), failure etc.

### PR DESCRIPTION
**Pull request by Dagonite**
Closes #106, closes #105, closes #104, closes #73.

## Description

Updates the reduction history page to be more fluid by using a collapsible table.

## How to test

Set the fronted end to use the live API and then run through SciGateway. Navigate to the reduction history page and then browse runs for the MARI instrument.

### CODE REVIEW

**Review by Pasarus — CHANGES_REQUESTED** [review]
This looks phenomenal and exactly how I envisioned it.

Only a couple issues that should be easily resolvable:
- When sorting by title the application seems to crash, we don't need to sort by title so feel free to turn this off.
- Can we swap the reduction start/end with the run start/end so the reduction details are in the drop down and the run details are in the main table?
- In the event that a run fails, it still says "Reduction outputs" where it should say stacktrace output (Even though we don't presently record this data fully)
- Could we give the values in the Run Details and Reduction Inputs a font-weight: 500; to make them "slightly" more bold than normal to reduce "noise"?
- Before merging I'd like to disable the View button until the feature is implemented, could both buttons get a tooltip saying "Work in progress..." or "Feature coming soon"?

Really happy with this so far, great work!

**Comment by Dagonite** [comment]
>When sorting by title the application seems to crash, we don't need to sort by title so feel free to turn this off.

Yes I had assumed sorting by title would be added to the API later but I'll remove it if it's not needed.

>Can we swap the reduction start/end with the run start/end so the reduction details are in the drop down and the run details are in the main table?

Yes will do.

>In the event that a run fails, it still says "Reduction outputs" where it should say stacktrace output (Even though we don't presently record this data fully)

Yeah I'll change this.

>Could we give the values in the Run Details and Reduction Inputs a font-weight: 500; to make them "slightly" more bold than normal to reduce "noise"?

Will do.

>Before merging I'd like to disable the View button until the feature is implemented, could both buttons get a tooltip saying "Work in progress..." or "Feature coming soon"?

Yes. And currently have a tooltip in place for the disabled "Download" button.

**Review by Pasarus — CHANGES_REQUESTED** [review]
Would it be possible to move this here?
image

Only other thing that I'd like to comment on/change is that it doesn't seem obvious to a user that the reduction inputs is scrollable notable problem on MARI, not sure if there is a good solution to this?

File: src/ReductionHistory.tsx [hunk 1]
```diff
@@ -198,120 +167,334 @@ const ReductionHistory: React.FC = () => {
...
+                    Reduction ID: {reduction.id}
+                  </Typography>
+                  <Typography variant="body2" gutterBottom sx={{ fontWeight: 'bold' }}>
+                    Instrument: {reduction.runs[0].instrument_name}
+                  </Typography>
+                  <Typography variant="body2" gutterBottom sx={{ fontWeight: 'bold' }}>
+                    Reduction start: {formatDateTime(reduction.reduction_start)}
+                  </Typography>
+                  <Typography variant="body2" gutterBottom sx={{ fontWeight: 'bold' }}>
+                    Reduction end: {formatDateTime(reduction.reduction_start)}
```
[the remark this answers is not in the document]
Comment (Dagonite) [reply src/ReductionHistory.tsx:454]: Oopsy

**Review by Pasarus — APPROVED** [review]
Thank you for this! It looks really good.

**Pull request merged**